aws/data_source_aws_sfn_activity.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sfn"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsSfnActivity() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsSfnActivityRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"creation_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsSfnActivityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient)
+	conn := client.sfnconn
+	log.Print("[DEBUG] Reading Step Function Activity")
+
+	params := &sfn.DescribeActivityInput{}
+
+	var name string
+	if nm, ok := d.GetOk("name"); ok {
+		name = nm.(string)
+		params.ActivityArn = aws.String(buildSfnActivityArn(name, client.accountid, client.region))
+	}
+
+	var arn string
+	if rnm, ok := d.GetOk("arn"); ok {
+		arn = rnm.(string)
+		params.ActivityArn = aws.String(arn)
+	}
+
+	act, err := conn.DescribeActivity(params)
+	if err != nil {
+		return err
```

Comment:
We should return error context here for operators and code maintainers:

```suggestion
		return fmt.Errorf("error describing SFN Activity (%s): %w", aws.StringValue(params.ActivityArn), err)
```